lib/middleman.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const reusify = require('reusify')
+
+function middleman (complete) {
+  var functions = []
+  var hasMiddlewares = false
+  var pool = reusify(Holder)
+
+  return {
+    use,
+    run
+  }
+
+  function use (f) {
+    hasMiddlewares = true
+    functions.push(f)
+    return this
+  }
+
+  function run (req, res) {
+    if (!hasMiddlewares) {
+      complete(null, req, res)
+      return
+    }
+
+    const holder = pool.get()
+    holder.req = req
+    holder.res = res
+    holder.done()
+  }
+
+  function Holder () {
+    this.next = null
```

Comment:
it's part of `reusify`, check the readme and the source there. It's defined there so it can be opimized: https://github.com/mcollina/reusify/blob/master/reusify.js#L11.